src/components/ProductPage.tsx

```diff
@@ -1,30 +1,24 @@
-import React, { useEffect, useState } from 'react'
 import ky from 'ky'
+import { useQuery } from 'react-query'
 import './../css/product.css'
 
-interface ProductPageProps {
+type ProductPageProps = {
   onAddToCart: (product: ProductItem) => void
 }
 
 function ProductPage({ onAddToCart }: ProductPageProps) {
-  const [product, setProduct] = useState<ProductItem | null>(null)
-  const productId = window.location.pathname.split('/product/')[1]
+  const productQuery = useQuery('product', fetchProduct)
```

Comment:
Here's an example of a custom hook where we can pass in the needed data to make the API call. Also React-Router has useful hooks to extract variables from the url
```suggestion
  const { productId } = useParams()
  
  const productQuery = useProductQuery({ productId })
```